src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -556,7 +554,7 @@ export const getTopNavConfig = (
               }
             ),
             testId: 'visualizesaveAndReturnButton',
-            disableButton: hasUnappliedChanges || !dashboardCapabilities.showWriteControls,
+            disableButton: hasUnappliedChanges,
```

Comment:
Why are we removing this check? `dashboardCapabilities.showWriteControls`

Reply:
I'm actually not going to include those changes in this PR and will address is a separate PR https://github.com/elastic/kibana/pull/121905, and we can move discussions over to that PR. I'm removing this check because it disables the `Save and return` button when you are trying to create a new vis from Canvas on a user with access to vis, lens, maps, etc, but without write access to dashboard

I don't believe this check is necessary since the presentation apps, i.e. Dashboard and Canvas handle disabling actions for authoring in an editor when a user does not have edit permissions. It makes more sense for the presentation apps to handle not directing to editors instead having the editors check presentation app permissions.